TACC/Core-Styles
# feat: s-form (and refactor c-form and django-cms-forms)

## Overview / Changes

1. Add `s-form` as a scope-version of `c-form`. 
2. Refactor `c-form` to style: `c-form`, `s-form`, `django-cms-forms`.
3. Use one form template for all form demos.

## Related

- [WP-63](https://jira.tacc.utexas.edu/browse/WP-63)
- creates `s-form` from https://github.com/TACC/tup-ui/pull/270
- expected by #205

## Testing

1. Verify all CMS form demos are identical **and** have CMS-styled errors:
    - http://localhost:3000/components/detail/c-form--cms
    - http://localhost:3000/components/detail/s-form--cms
    - http://localhost:3000/components/detail/django-cms-forms
    <sub>Remember to "Toggle Field Errors" (button at bottom of form).</sub>
1. Verify all Portal form demos are identical **and** have Portal-styled errors:
    - http://localhost:3000/components/detail/c-form--portal
    - http://localhost:3000/components/detail/s-form--portal
    <sub>Remember to "Toggle Field Errors" (button at bottom of form).</sub>
1. Verify all Default form demos are identical **and** have un-styled errors:
    - http://localhost:3000/components/detail/c-form--default
    - http://localhost:3000/components/detail/s-form--default
    <sub>The error messages are not styled, because clients have their own styles.</sub>
1. Verify all form demos support toggling required fields:
    - Upon toggle **off**:
    	- The first "(required)" field will become "(important)".
    	- That first field will **not** be required to submit the form.
    	- The second "(required)" field will _remain_ "(required)".
    	    <sub>It is not a real field. It is JavaScript Captcha. It can not be un-required.</sub>
    - Upon toggle **on**:
    	- The first "(important)" field will become "(required)".
    	- That first field **will** be required to submit the form.
    	- The second "(required)" field will _remain_ "(required)".

## UI

https://github.com/TACC/Core-Styles/assets/62723358/b00b699c-31bf-4aeb-9c44-a39a6407bd1f
